content/client-lib-development-guide/features.textile

```diff
@@ -287,9 +292,10 @@ h3(#realtime-connection). Connection
 * @(RTN6)@ A @Connection@ is successful and considered @CONNECTED@ once the websocket connection is open and the intial @CONNECTED@ @ProtocolMessage@ has been received
 * @(RTN19)@ If the @CONNECTED@ @ProtocolMessage@ contains a @connectionDetails@ property, the attributes within @ConnectionDetails@ will be used as the defaults for this client library, overriding any configured options at the time the @CONNECTED@ @ProtocolMessage@ is received
 * @(RTN7)@ @ACK@ and @NACK@:
-** @(RTN7a)@ All @ProtocolMessage@ @Presence@ and @Message@ objects sent to Ably expect either an @ACK@ or @NACK@ from Ably to confirm successful receipt and acceptance or failure respectively
+** @(RTN7a)@ All @ProtocolMessage@ @Presence@ and @Message@ objects sent to Ably expect either an @ACK@ or @NACK@ from Ably to confirm successful receipt and acceptance or failure respectively. For clarity, it is unnecessary to fail the publish operation of a message using a timer. Instead the client library can rely on the realtime system to send an @ACK@ or @NACK@ when connected, to fail all awaiting messages when @DISCONNECTED@ for too long as described in "RTN7d":#RTN7d, or upon reconnecting, the corresponding @ACK@ or @NACK@ protocol messages will be received as described in "RTN19a":#RTN19a
 ** @(RTN7b)@ Every @ProtocolMessage@ sent must contain a unique serially incrementing @msgSerial@ integer value starting at zero. The @msgSerial@ along with the @count@ for incoming @ACK@ and @NACK@ @ProtocolMessages@ indicates which messages succeeded or failed to be delivered
 ** @(RTN7c)@ If a connection enters the @CLOSED@ or @FAILED@ state, or if the connection state is lost, and a @ACK@ or @NACK@ has not yet been received for a message, the client should consider the delivery of those messages as failed
+** @(RTN7d)@ If a connection enters the @DISCONNECTED@ or @SUSPENDED@ state, and has not become @CONNECTED@ within 15 seconds, then the client should consider the delivery of those messages as failed
```

Comment:
> if we failed all messages already sent to the realtime system, then it's quite plausible devs will be told their message send failed and will send again, only to discover it was published twice. By resending the original messages on the new transport the realtime system prevents that from happening. 

@mattheworiordan I'm sorry, I'm getting really confused here. AFAICT, this is an argument against nacking-after-15s for messages that have already been attempted once, and in favour of nacking-after-15s for messages which haven't been attempted yet. But the spec item (RTN7d) we're commenting on only applies to messages which *have* been attempted (RTN7) and are awaiting ack/nack.